src/sparseml/pytorch/torchvision/train.py

```diff
@@ -430,6 +454,15 @@ def collate_fn(batch):
         if scaler:
             scaler.load_state_dict(checkpoint["scaler"])
 
+        # NOTE: override manager with the checkpoint's manager
+        manager = ScheduledModifierManager.from_yaml(checkpoint["checkpoint_recipe"])
+        checkpoint_manager = None
+
+        # TODO do we still do this?
+        manager.apply_structure(model_without_ddp, epoch=checkpoint["epoch"])
```

Comment:
instead should `initialize` the manager instead of `apply_structure` so that we get to the exact model and manager state at the given epoch. (apply structure involves calling into both `initialize` and `finalize`)

Reply:
after the manager is initialized and the model object structure is recovered, it's safe to call into `model.load_state_dict`, not safe before on resume

Reply:
Updated - check if new logic is correct

Reply:
looks good